app/app.go

```diff
@@ -191,3 +188,9 @@ func splitKey(key string) (prefix string, suffix string) {
 	}
 	return key, ""
 }
+
+// (not meant to be called)
```

Comment:
oy I think I was caught by this kind of thing once before. Thanks for adding this check.

Maybe a simpler way is `var _ = BlockchainAware(new(Basecoin))`

Alternatively, I'm not sure if there's actually any value in an app that doesnt implement `BlockchainAware` ... maybe we can just make that part of the interface. @jaekwon ?